WoodStar/WoodStar/TdsChannel.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Buffers;
+using System.IO;
+using System.IO.Pipelines;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+
+namespace WoodStar
+{
+    public class TdsChannel
+    {
+        private readonly PipeReader _pipeReader;
+        private readonly Stream _stream;
+
+        public TdsChannel(Stream stream)
+        {
+            _pipeReader = PipeReader.Create(stream);
+            _stream = stream;
+        }
+
+        public async Task Prelogin()
+        {
+            var preloginStream = new PreloginStream(0, 0, 1, 0, EncryptionOption.Off, null, 0, false, null, null, null);
+            var header = new TdsHeader(PacketType.PreLogin, PacketStatus.EOM, preloginStream.Length + TdsHeader.HeaderSize, 0, 1);
+            var preloginBuffer = ArrayPool<byte>.Shared.Rent(header.Length);
+            header.WriteToBuffer(preloginBuffer);
+            preloginStream.WriteToBuffer(preloginBuffer);
+
+            await _stream.WriteAsync(preloginBuffer, 0, header.Length);
+
+            ArrayPool<byte>.Shared.Return(preloginBuffer);
+
+            var preloginResponse = await ReadNextPacketAsync(ResponseType.Prelogin);
+            if (preloginResponse is PreloginPacket preloginPacket)
+            {
+                // Assign version etc info to connection
+            }
+        }
+
+        public async Task Login()
+        {
+            var login7Stream = new Login7Stream
+            {
+                UserName = "sa",
+                Password = "Password1!"
+            };
+            var header = new TdsHeader(PacketType.Login, PacketStatus.EOM, login7Stream.Length + TdsHeader.HeaderSize, 0, 1);
+            var loginBuffer = ArrayPool<byte>.Shared.Rent(header.Length);
+            header.WriteToBuffer(loginBuffer);
+            login7Stream.WriteToBuffer(loginBuffer);
+
+            await _stream.WriteAsync(loginBuffer, 0, loginBuffer.Length);
```

Comment:
Pretty sure it's supposed to be `header.Length` just like above, instead of `loginBuffer.Lengt`.